cmd/admin-handlers.go

```diff
@@ -68,6 +70,72 @@ func (adminAPI adminAPIHandlers) ServiceRestartHandler(w http.ResponseWriter, r
 	sendServiceCmd(globalAdminPeers, serviceRestart)
 }
 
+// setCredsReq request
+type setCredsReq struct {
+	Username string `xml:"username"`
+	Password string `xml:"password"`
+}
+
+// ServiceCredsHandler - POST /?service
+// HTTP header x-minio-operation: creds
+// ----------
+// Update credentials in a minio server. In a distributed setup, update all the servers
+// in the cluster.
+func (adminAPI adminAPIHandlers) ServiceCredentialsHandler(w http.ResponseWriter, r *http.Request) {
+	// Authenticate request
+	adminAPIErr := checkRequestAuthType(r, "", "", "")
+	if adminAPIErr != ErrNone {
+		writeErrorResponse(w, adminAPIErr, r.URL)
+		return
+	}
+
+	// Avoid setting new credentials when they are already passed
+	// by the environnement
+	if globalEnvAccessKey != "" || globalEnvSecretKey != "" {
+		writeErrorResponse(w, ErrMethodNotAllowed, r.URL)
+		return
+	}
+
+	// Load request body
+	inputData, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		writeErrorResponse(w, ErrInternalError, r.URL)
+		return
+	}
+
+	// Unmarsh request body
+	var req setCredsReq
+	err = xml.Unmarshal(inputData, &req)
+	if err != nil {
+		errorIf(err, "Cannot unmarshal credentials request")
+		writeErrorResponse(w, ErrMalformedXML, r.URL)
+		return
+	}
+
+	// Check passed credentials
+	cred, err := getCredential(req.Username, req.Password)
+	if err != nil {
+		writeErrorResponse(w, ErrAdminBadCred, r.URL)
```

Comment:
> Do we need a new error value ErrAdminBadCred? getCredential would return one of errInvalidAcessKeyLength or errInvalidSecretKeyLength. They would be more specific than ErrAdminBadCred, i.e, whether the accesskey is incorrect or the secretkey is incorrect (or both).

errInvalidAcessKeyLength and errInvalidSecretKeyLength are standard errors, we need an ErrorResponse type which represents the regular S3 xml error like:
```
 27 <?xml version="1.0" encoding="UTF-8"?>
 28 <Error>
 29    <Code>AccessDenied</Code>
 30    <Message>Access Denied</Message>
 31    <BucketName>bucketName</BucketName>
 32    <Key>objectName</Key>
 33    <RequestId>F19772218238A85A</RequestId>
 34    <HostId>GuWkjyviSiGHizehqpmsD1ndz5NClSP19DOT+s2mv7gXGQ8/X1lhbDGiIJEXpGFD</HostId>
 35 </Error>
```